### PR TITLE
Candybar: VIA support for lefty and righty

### PR DESCRIPTION
**Pull request by TerryMathews**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/candybar/lefty/config.h [hunk 1]
```diff
@@ -19,11 +19,11 @@
...
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x6060
+#define VENDOR_ID       0x544B //TK
+#define PRODUCT_ID      0x0003
 #define DEVICE_VER      0x0006
 #define MANUFACTURER    The Key Company
-#define PRODUCT         Candybar
+#define PRODUCT         The Key Company Candybar
```
Comment (fauxpark) [inline keyboards/candybar/lefty/config.h:26]:
```suggestion
#define PRODUCT         Candybar
```

File: keyboards/candybar/righty/config.h [hunk 2]
```diff
@@ -19,11 +19,11 @@
...
 
 /* USB Device descriptor parameter */
-#define VENDOR_ID       0xFEED
-#define PRODUCT_ID      0x6060
+#define VENDOR_ID       0x544B //TK
+#define PRODUCT_ID      0x0002
 #define DEVICE_VER      0x0006
 #define MANUFACTURER    The Key Company
-#define PRODUCT         Candybar
+#define PRODUCT         The Key Company Candybar
```
Comment (fauxpark) [inline keyboards/candybar/righty/config.h:26]:
```suggestion
#define PRODUCT         Candybar
```

**Comment by TerryMathews** [comment]
Glad to make this change, however it conflicts with the VIA porting instructions.

From https://caniusevia.com/docs/configuring_qmk/

Change PRODUCT
The text defined by the PRODUCT symbol in config.h is what will appear in the list of devices (for example, in the ‘Bluetooth & other devices’ page of Windows, and in a notification when the device is first connected and being ‘installed’).

It is highly recommended that the value of PRODUCT is changed to be a combination of designer/vendor and the device name. For example:

#define MANUFACTURER    wilba.tech
#define PRODUCT         wilba.tech WT60-D
#define DESCRIPTION     wilba.tech WT60-D
VIA Configurator will in future switch to using the value of PRODUCT when displaying the device’s name, rather than the name in the <keyboard_name>.json. This will allow firmware level customization.

Note that spaces are allowed.

Please advise.

**Comment by fauxpark** [comment]
I strongly disagree with those instructions, because it is *only* a Windows thing. Pretty much everywhere else, including the Toolbox, it will show up as eg. `wilba.tech wilba.tech WT60-D`.

**Comment by TerryMathews** [comment]
Changes committed

File: keyboards/candybar/lefty/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  KC_LSFT,KC_Z,KC_X,KC_C,KC_V,KC_B,KC_N,KC_M,KC_COMM,KC_DOT,KC_RSFT,KC_UP,KC_P1,KC_P2,KC_P3,KC_PPLS, \
+  KC_LCTL,KC_LGUI,KC_LALT,KC_SPC,KC_SPC,KC_BSPC,KC_APP,MO(_FL),KC_LEFT,KC_DOWN,KC_RGHT,KC_P0,KC_PDOT,KC_PENT),
+
+  /* Keymap _FL: Function Layer
+   */
+[_FL] = LAYOUT(
+  RESET,KC_Q,KC_W,KC_E,KC_R,KC_T,KC_Y,KC_U,KC_I,KC_LBRC,KC_RBRC,KC_INS,KC_BSPC,KC_P7,KC_P8,KC_P9,KC_VOLU, \
+  KC_TAB,KC_A,KC_SLCK,KC_D,KC_F,KC_G,KC_H,KC_J,KC_K,KC_L,KC_QUOT,KC_BSLS,KC_P4,KC_P5,KC_P6,KC_VOLD, \
+  KC_LSFT,KC_Z,KC_X,KC_CAPS,KC_V,KC_B,KC_NLCK,KC_M,KC_COMM,KC_DOT,KC_SLSH,KC_PGUP,KC_P1,KC_P2,KC_P3,KC_PEQL, \
+  KC_LCTL,KC_LGUI,KC_LALT,KC_SPC,KC_SPC,KC_BSPC,KC_APP,MO(_FL),KC_HOME,KC_PGDN,KC_END,KC_P0,KC_PDOT,KC_PENT),
```
Comment (zvecr) [inline keyboards/candybar/lefty/keymaps/via/keymap.c:37]: Looks like the wrong keymap got duplicated.

```suggestion
  /* Keymap _BL: Base Layer (Default Layer)
   */
[_BL] = LAYOUT(
  KC_P7  , KC_P8, KC_P9  , KC_PAST, KC_ESC , KC_Q   , KC_W   , KC_E, KC_R   , KC_T  , KC_Y, KC_U   , KC_I   , KC_O   , KC_P   , KC_DEL , KC_BSPC , \
  KC_P4  , KC_P5, KC_P6  , KC_PMNS, KC_TAB , KC_A   , KC_S   , KC_D, KC_F   , KC_G  , KC_H, KC_J   , KC_K   , KC_L   , KC_SCLN,          KC_ENT  , \
  KC_P1  , KC_P2, KC_P3  , KC_PPLS, KC_LSFT,          KC_Z   , KC_X, KC_C   , KC_V  , KC_B, KC_N   , KC_M   , KC_COMM, KC_DOT , KC_UP   ,KC_RSFT , \
  MO(_FL), KC_P0, KC_PDOT, KC_PENT, KC_LCTL, KC_LGUI, KC_LALT,                KC_SPC,       KC_SPC , KC_BSPC, KC_APP , KC_LEFT, KC_DOWN, KC_RGHT),

  /* Keymap _FL: Function Layer
  */
  [_FL] = LAYOUT(
  KC_P7 , KC_P8, KC_P9  , KC_VOLU, RESET  , KC_Q   , KC_W   , KC_E, KC_R   , KC_T  , KC_Y, KC_U   , KC_I   , KC_LBRC, KC_RBRC, KC_INS , KC_BSPC , \
  KC_P4 , KC_P5, KC_P6  , KC_VOLD, KC_TAB , KC_A   , KC_SLCK, KC_D, KC_F   , KC_G  , KC_H, KC_J   , KC_K   , KC_L   , KC_QUOT,          KC_BSLS , \
  KC_P1 , KC_P2, KC_P3  , KC_PEQL, KC_LSFT,          KC_Z   , KC_X, KC_CAPS, KC_V  , KC_B, KC_NLCK, KC_M   , KC_COMM, KC_DOT , KC_SLSH, KC_PGUP , \
  KC_END, KC_P0, KC_PDOT, KC_PENT, KC_LCTL, KC_LGUI, KC_LALT,                KC_SPC,       KC_SPC , KC_BSPC, KC_APP , MO(_FL), KC_HOME, KC_PGDN),
```

**Comment by Wilba6582** [comment]
> I strongly disagree with those instructions, because it is _only_ a Windows thing. Pretty much everywhere else, including the Toolbox, it will show up as eg. `wilba.tech wilba.tech WT60-D`.

It is not _only_ a Windows thing. OSX does _exactly the same thing_

image

So the two major operating systems (used by 95% of users) do it this way. "Everywhere else" being Linux's `lsusb` command and QMK Toolbox concatenate manufacturer and product, and the latter only if a feature is enabled.

File: keyboards/candybar/lefty/keymaps/via/keymap.c (hunk 3 above)
Comment (TerryMathews) [reply keyboards/candybar/lefty/keymaps/via/keymap.c:37]: Nice catch!

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**